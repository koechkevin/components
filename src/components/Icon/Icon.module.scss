@import '../../styles/global';

.icon {
  width: 32px;
  height: 32px;
  min-width: 32px;
  line-height: 32px;
  text-align: center;
  border-radius: 2px;
  user-select: none;
  color: inherit;
  @include flex-center;
  display: inline-flex;

  &.hover {
    cursor: pointer;
  }

  &.hover:hover {
    background-color: $color-basic-light-gray;
  }

  svg {
    width: 16px;
    height: 16px;
    font-size: 16px;
  }
}
