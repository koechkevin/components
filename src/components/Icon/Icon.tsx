import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
import React, { CSSProperties, FC } from 'react';

import styles from './Icon.module.scss';

interface Props extends FontAwesomeIconProps {
  id?: string;
  hover?: boolean;
  style?: CSSProperties;
  className?: string;
  onClick?: (e: any) => void;
}

const Icon: FC<Props> = (props) => {
  const { id, style, hover, className, onClick, ...restProps } = props;
  const cls = [styles.icon, hover ? styles.hover : '', className || ''].join(' ');

  return (
    <span onClick={(e) => onClick && onClick(e)} id={id} style={style} className={cls}>
      <FontAwesomeIcon {...restProps} />
    </span>
  );
};

export default Icon;
