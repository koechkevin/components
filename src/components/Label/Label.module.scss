@import '../../styles/global.scss';

.label {
  @include flex-center-vertical;
  height: 32px;
  font-size: 12px;
  margin-right: 8px;
  font-weight: 600;
  line-height: 18px;
  color: $color-primary;
  background-color: $color-primary-light;
}
