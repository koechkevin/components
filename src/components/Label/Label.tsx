import React, { FC } from 'react';

import styles from './Label.module.scss';

export interface Props {
  leading?: React.ReactNode;
  trailing?: React.ReactNode;
  className?: string;
  color?: string;
  background?: string;
  style?: React.CSSProperties;
}

const Label: FC<Props> = (props) => {
  const { children, style, className, color, ...restProps } = props;
  const classNames: string = [styles.label, className ? className : ''].join(' ');

  return (
    <span
      style={{
        color: props.color,
        height: 32,
        borderRadius: 4,
        background: props.background,
        paddingLeft: props.leading ? 0 : 8,
        paddingRight: props.trailing ? 0 : 8,
        ...style,
      }}
      className={classNames}
      {...restProps}
    >
      {props.leading && props.leading}
      {children}
      {props.trailing && props.trailing}
    </span>
  );
};

export default Label;
