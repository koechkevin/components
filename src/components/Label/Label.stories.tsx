import { faCheck, faSquare, faTimes} from '@fortawesome/pro-regular-svg-icons';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Center } from '../Center/Center';
import { Icon } from '../Icon';
import Label from './Label';

storiesOf('Label', module)
  .addDecorator((story: any) => <Center>{story()}</Center>)
  .add('default', () => <Label>Choice Label</Label>)
  .add('trailing icon', () => <Label trailing={<Icon icon={faTimes} />}>Choice Label</Label>)
  .add('Double Icon', () => (
    <Label trailing={<Icon icon={faSquare} />} leading={<Icon icon={faSquare} />}>
      Double Icon
    </Label>
  ))
  .add('With leading only', () => <Label leading={<Icon icon={faCheck} />}>Choice Label</Label>)
  .add('with custom colors', () => (
    <>
      <Label color="#2b9037" background="#dfece1">
        Recruiter
      </Label>
      <Label color="#9c43d6" background="#eedff8">
        Billing
      </Label>
      <Label color="#d7a84b" background="#faeccb">
        Owner
      </Label>
      <Label color="#565656" background="#eaeaea">
        Guest
      </Label>
      <Label color="#565656" background="#eaeaea">
        Team Member
      </Label>
    </>
  ));
