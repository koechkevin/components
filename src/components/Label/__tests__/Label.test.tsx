import { faSquare } from '@fortawesome/pro-regular-svg-icons';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';

import { Icon } from '../../Icon';
import Label, { Props } from '../Label';

describe('Tests for label', () => {
  const props: Props = {
    leading: <Icon icon={faSquare} />,
    trailing: <Icon icon={faSquare} />,
  };
  const text: string = 'Text';
  const wrapper: ReactWrapper = mount(<Label {...props}>{text}</Label>);

  it('Should mount without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should contain the label Text', () => {
    expect(wrapper.text()).toEqual(text);
  });

  it('Should have trailing and leading elements', () => {
    const leading: ReactWrapper = wrapper.prop('leading');
    const trailing: ReactWrapper = wrapper.prop('trailing');

    expect(leading).toEqual(props.leading);
    expect(trailing).toEqual(props.trailing);
  });
});
