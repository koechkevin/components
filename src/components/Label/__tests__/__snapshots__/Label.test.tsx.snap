// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for label Should mount without crushing 1`] = `ReactWrapper {}`;
